docker-compose.yml

```diff
@@ -30,3 +35,46 @@ services:
             - 27017:27017
         logging:
             driver: none
+
+    mongo1:
```

Comment:
Are all of these instances required to build transactions?

Reply:
Transaction requires mongodb version V4.0 or more and deployment replica sets or sharded clusters, so i must create this environment

Reply:
So mongo has no support for transactions with rollbacks if you're running only 1 instance?

Reply:
That might be confusing for people coming from SQL setups.

Reply:
It will report error when you execute `beginTransaction` if the environment is not supported

Reply:
Error [here](https://github.com/Smolevich/demo-mongo-application/pull/9) : `Transaction numbers are only allowed on a replica set member or mongos`

Reply:
> So mongo has no support for transactions with rollbacks if you're running only 1 instance?

Exactly, I found out about that the hard way, barely found it mentioned in the documentation

Reply:
see [documentation](https://docs.mongodb.com/manual/core/transactions/#transactions-api)

> - For transactions on MongoDB 4.2 deployments (replica sets and sharded clusters), clients must use MongoDB drivers updated for MongoDB 4.2.
> - When using the drivers, each operation in the transaction must be associated with the session (i.e. pass in the session to each operation).

Reply:
> So mongo has no support for transactions with rollbacks if you're running only 1 instance?

Late to the discussion here, but I'll point out that you can always create a single-member replica set for testing (not recommended for production). Features such as transactions and change streams just depend on there being oplog, which will exist even with a lone primary server.